# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install fb_mandelbrot $DESTDIR/fb/bin
install fb_rectangle $DESTDIR/fb/bin
install fb_sierpinski $DESTDIR/fb/bin

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/fbmark-build.pc << EOF
Name: fbmark
Version: 0.3
Description: Linux Framebuffer Benchmark
Requires: $REQUIRES

exec=\\
/fb/bin/fb_mandelbrot \\
/fb/bin/fb_rectangle \\
/fb/bin/fb_sierpinski
EOF

exit

# patch
--- fbmark-0.3.orig/Configure
+++ fbmark-0.3/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile

# source
https://github.com/caramelli/fbmark/archive/v0.3/fbmark-0.3.tar.gz
